### PR TITLE
Features/BQT-2578 stripe sca

### PR DESCRIPTION
**Pull request by str1ke**

### CODE REVIEW

File: src/providers/stripe_sca.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+      } else {
+        alert(message);
+      }
+    } else {
+      const action = gatewaySettings.bookingToken
+        ? preparePaymentIntentForBooking(gatewaySettings.bookingToken, result.paymentMethod.id)
+        : prepareIntent(gatewaySettings.customer_data.amount, gatewaySettings.apiKey, result.paymentMethod.id);
+
+        action
+        .then((response) => {
```
Comment (str1ke) [inline src/providers/stripe_sca.js]: indentation

File: src/providers/stripe_sca.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+              result.paymentMethod.card.last4,
+              undefined, // TODO: put cardholder name here
+              {
+                clientSecret: response.client_secret
+              }
+            );
+          }
+
+          hideForm();
+        });
```
Comment (str1ke) [inline src/providers/stripe_sca.js]: need to add catch

File: src/providers/stripe_sca.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+          return response.json();
+        })
+        .then((response) => {
+          if (gatewaySettings.onTokenize && typeof(gatewaySettings.onTokenize) === 'function') {
+            // TODO: do we need data here for online payment gateways?
+            gatewaySettings.onTokenize(
+              result.paymentMethod.id,
+              result.paymentMethod.card.last4,
+              undefined, // TODO: put cardholder name here
```
Comment (str1ke) [inline src/providers/stripe_sca.js]: get actual cardholder name here

File: src/providers/stripe_sca.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      amount: amount,
+      payment_method_id: paymentMethodId
+    })
+  });
+}
+
+function onSubmit(event) {
+  event.preventDefault();
+
+  stripe.createPaymentMethod('card', card).then(function(result) {
```
Comment (str1ke) [inline src/providers/stripe_sca.js]: need to check what billing information contains payment method and what information we have on form

File: src/online_providers/stripe_sca.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+export default {
+  handleOnlinePayment: ({ settings, payment }) => {
+    const token = settings.token;
+    const clientSecret = payment.clientSecret;
+
```
Comment (Judis) [inline src/online_providers/stripe_sca.js]: Redundant whitespace?

File: src/providers/stripe_sca.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
+const libraryPaths = [
+  'https://js.stripe.com/v3/'
+];
+const DOMAIN = "https://app.thebookingfactory.com";
```
Comment (Judis) [inline src/providers/stripe_sca.js]: Remember about WhiteLabel solutions.
Don't hardcode TheBookingFactory domain name.

Comment (str1ke) [reply src/providers/stripe_sca.js]: add same approach for domain configuration as in public-booking-widget

File: src/online_providers/stripe_sca.js (hunk 5 above)
Comment (str1ke) [reply src/online_providers/stripe_sca.js]: It have a meaning to separate logical parts of method - prepare/process input, act, return result. This is part of preparing data to be used in method, i.e its dependencies.

**Pull request merged**